kimchi/src/circuits/gate.rs

```diff
@@ -235,6 +235,7 @@ impl<F: FftField> LookupInfo<F> {
 
     /// Each entry in `kinds` has a corresponding selector polynomial that controls whether that
     /// lookup kind should be enforced at a given row. This computes those selector polynomials.
+    #[allow(clippy::type_complexity)]
```

Comment:
Is there a way to make these returned types clearer? I think Clippy is right to warn here.